parallel/slice.go

```diff
@@ -1,134 +1,88 @@
 package parallel
 
-import "sync"
+import (
+	"github.com/samber/lo/optional"
+	"runtime"
+	"sync"
+)
 
-// Map manipulates a slice and transforms it to a slice of another type.
-// `iteratee` is call in parallel. Result keep the same order.
-func Map[T any, R any](collection []T, iteratee func(T, int) R) []R {
-	result := make([]R, len(collection))
-
-	var wg sync.WaitGroup
-	wg.Add(len(collection))
-
-	for i, item := range collection {
-		go func(_item T, _i int) {
-			res := iteratee(_item, _i)
+var DefaultPoolSize = runtime.NumCPU() / 2
```

Comment:
Is it `runtime.NumCPU()` can be 1, then `DefaultPoolSize` will be zero?

Reply:
I know some use cases will cause confusing behavior about default pool size. 

Let says someone wants to request 10 URLs in parallel on a device had 2 CPUs, he will write some code just like that:

```go
responses := parallel.Map(URLs, request)
```

In this case, requesting a URL is only an IO-intensive task, so for the best performance and efficiency, it should requesting 10 URLs at the same time, just like the code above looks like for the package user.  

But DefaultPoolSize is working internally and it is `1` (half of the CPU number), so actually, only one requesting task is started at the same time. And in this case, someone will confused about **why the code will be slower 10x times**.

The task pool can improve performance only if the developer knows what they doing, and adjusts the pool size thoughtfully for the special scene. A global default pool size can't improve performance in any scene, and often it's worse (if it can, the Golang GMP will make it internally).

I recommend removing the default pool size, and only using the task pool when the concurrency limit option is set.